File: erlang/p_16.erl
-module(p_16).
-export([drop_nth/2]).
-import(take_drop, [take/2, drop/2]).
-import(lists, [append/2]).

drop_nth(_, N) when N < 0 ->
    erlang:error(negative);
drop_nth([], _) -> [];
drop_nth(Xs, N) ->
    append(take(Xs, N-1), drop_nth(drop(Xs, N), N)).

File: prolog/p-16.pl
:- ensure_loaded('take-drop').

drop_every([], _, []).
drop_every(Xs, N, Ys) :-
    N > 0, N1 is N - 1,
    take(N1, Xs, P1),
    drop(N, Xs, P),
    drop_every(P, N, P2),
    append(P1, P2, Ys), !.
